zipp.py

```diff
@@ -220,9 +260,32 @@ def __init__(self, root, at=""):
         original type, the caller should either create a
         separate ZipFile object or pass a filename.
         """
-        self.root = FastLookup.make(root)
+        self.root = self._make_root(root)
         self.at = at
 
+    def _make_root(self, source):
```

Comment:
I'm not happy that this function appears to copy functionality of `CompleteDirs.make()`, tweak it, and abandon the old code. More concerning is that it brings together a lot of concerns (Pickleability, CompleteDirs, FastLookup, mutating the source class), all outside of those classes.

I'm not sure what solution I'd like here, but I'd like to address these concerns. There's some good inspiration here (especially in that it works), so I'll use it and see if I can address some of these concerns.

Reply:
The latest implementation no longer needs `_make_root` and continues to use `FastPath.make`.